### PR TITLE
LIMS-7447: Make sure that user can filter by sub & super scripts in the filter drop down list

### PR DESCRIPTION
**Pull request by modeso-NahlaMakhlouf**
- filter in test unit field by super script& sub script values and make sure that items in drop down list includes search text 

```
 C:\Users\ADMIN\PycharmProjects\1lims-automation>nosetests` -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test068_filter_testunit
_by_scripts_1_super_script
C:\Users\ADMIN\AppData\Local\Programs\Python\Python38\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlu
gin: No module named '_curses'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-07-26 16:15:49.313 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-07-26 16:15:49.316 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-07-26 16:15:49.842 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : _wtY1mNPliMmIaxiQTQW7pK1wBvXeP2UP9PJi3FJo6A .....

DevTools listening on ws://127.0.0.1:61006/devtools/browser/251ef322-8a77-41d7-9e63-177fcafce3c6
LIMS-7447- Make sure that user can filter by sub & super scripts in the filter drop down list [with value='super_script'] ...
2020-07-26 16:16:29.971 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test068_filter_testunit_by_scripts_1_super_script
2020-07-26 16:16:37.801 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:562 - wait until page is loaded
2020-07-26 16:16:37.899 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-07-26 16:16:39.178 | INFO     | api_testing.apis.orders_api:set_configuration:377 - set order configuration
2020-07-26 16:16:48.699 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-07-26 16:16:48.705 | INFO     | api_testing.apis.test_unit_api:set_configuration:407 - set test unit configuration
2020-07-26 16:17:17.318 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-07-26 16:17:17.334 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-07-26 16:17:17.848 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-07-26 16:17:23.561 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:562 - wait until page is loaded
2020-07-26 16:17:23.659 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-07-26 16:17:25.168 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-07-26 16:17:28.921 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-07-26 16:17:48.434 | INFO     | ui_testing.testcases.base_test:tearDown:33 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 120.456s

OK
```

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -10,16 +10,16 @@ jobs:
...
 
     steps:
       - uses: actions/checkout@master
       - name: pull docker image
         run:  docker pull 0xislamtaha/seleniumchromenose:83
       - name: execute test case in parallel
         env:
-          TEST_FILE: ui_testing/testcases/basic_tests/test04_testunits.py
-          TEST_REG: test[0-9]{3}
+          TEST_FILE: ui_testing/testcases/basic_tests/test06_orders.py
```
[the remark this answers is not in the document]
Comment (modeso-NahlaMakhlouf) [reply .github/workflows/CI.yml:21]: done

File: requirements.txt [hunk 2]
```diff
@@ -1,13 +1,16 @@
 nose
 nose-progressive
-nose-testconfig
+nose-testconfig~=0.10
```
[the remark this answers is not in the document]
Comment (modeso-NahlaMakhlouf) [reply requirements.txt:3]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 3]
```diff
@@ -2296,3 +2300,68 @@ def test065_download_suborder_sheet_for_single_order(self):
                                   f"{str(fixed_sheet_row_data)} : {str(formatted_orders[index])}")
             for item in formatted_orders[index]:
                 self.assertIn(item, fixed_sheet_row_data)
+
+
+
+    @parameterized.expand(['cancel_btn', 'close_btn'])
+    def test066_close_testplan_popup(self,button):
```
[the remark this answers is not in the document]
Comment (modeso-NahlaMakhlouf) [reply ui_testing/testcases/basic_tests/test06_orders.py:2307]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 4]
```diff
@@ -2296,3 +2300,68 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+        self.order_page.set_new_order()
+        self.order_page.sleep_small()
+        self.info('Asset that archived contact is not existing in the list')
+        self.assertFalse(self.order_page.is_contact_existing(archived_contact))
+
+    @parameterized.expand(['sub_script', 'super_script'])
+    def test068_filter_testunit_by_scripts(self, value):
+        """ LIMS-7447- Make sure that user can filter by sub & super scripts in the filter drop down list
+                                        """
+        self.test_unit_api.set_configuration()
```
[the remark this answers is not in the document]
Comment (modeso-NahlaMakhlouf) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 5]
```diff
@@ -2296,3 +2300,40 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+             LIMS-7447
+        """
+
+        self.test_unit_api.set_name_configuration()
+        if value == 'sub_script':
+            api, testunit = self.test_unit_api.create_qualitative_testunit(unit='14[158]')
+            self.assertEqual(api['status'], 1)
+        else:
+            api, testunit = self.test_unit_api.create_qualitative_testunit(unit='15{158}')
+            self.assertEqual(api['status'], 1)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:2316]: you can move this command outside if else as it's common between if and else

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 6]
```diff
@@ -2296,3 +2300,40 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+
+        unit = testunit['unit']
+        self.orders_page.get_orders_page()
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:2319]: why you need to navigate to order page, while set up gets order page and you didn't navigated to another page in previous steps

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 7]
```diff
@@ -2296,3 +2300,40 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+        self.order_page.sleep_small()
+        self.base_selenium.scroll(False)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:2321]: why do you need to scroll her, when orders page is opened, it's already scrolled up

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 8]
```diff
@@ -2296,3 +2300,40 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+
+
+
+
+
+
+
+
+
+
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:2339]: please remove unneeded empty lines

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 9]
```diff
@@ -2296,3 +2300,40 @@ def test065_download_suborder_sheet_for_single_order(self):
...
+            api, testunit = self.test_unit_api.create_qualitative_testunit(unit='15{158}')
+            self.assertEqual(api['status'], 1)
+
+        unit = testunit['unit']
+        self.orders_page.get_orders_page()
+        self.order_page.sleep_small()
+        self.base_selenium.scroll(False)
+        self.orders_page.apply_filter_scenario(filter_element='orders:test_units_filter',
+                                               filter_text=unit, field_type='drop_down')
+        self.assertIsNotNone(self.base_selenium.is_text_included_in_drop_down_items(element='orders:test_units_filter', item_text=unit))
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:2324]: in Jira ticket of this test case, step 5:
you need to assert that result display.
So you will create order with this test unit then get filter result data and assert that unit found in it

**Review by ghada-37 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**